executor/include/execution/executor.hpp

```diff
@@ -7,3 +7,77 @@
 #include <execution/executor/inline_executor.hpp>
 #include <execution/executor/omp_executor.hpp>
 #include <execution/executor/sse_executor.hpp>
+
+TYPE_TO_STRING(inline_executor<oneway_t, single_t, blocking_t::always_t>);
+TYPE_TO_STRING(sse_executor<oneway_t, bulk_t, blocking_t::always_t>);
+TYPE_TO_STRING(omp_executor<oneway_t, bulk_t, blocking_t::always_t>);
+TYPE_TO_STRING(cuda_executor<oneway_t, bulk_t, blocking_t::always_t>);
+
+#define executors                                                \
+  std::tuple<                                                    \
+      inline_executor<oneway_t, single_t, blocking_t::always_t>, \
+      sse_executor<oneway_t, bulk_t, blocking_t::always_t>,      \
+      omp_executor<oneway_t, bulk_t, blocking_t::always_t>,      \
+      std::conditional<                                          \
+          execution::is_executor_available_v<cuda_executor>,     \
+          cuda_executor<oneway_t, bulk_t, blocking_t::always_t>, \
+          inline_executor<oneway_t, single_t, blocking_t::always_t>>::type>
+
+TEST_CASE_TEMPLATE_DEFINE("Validity ", E, validity) {
+  auto exec = E{};
+
+  SUBCASE("is_executor") {
+    CHECK(execution::is_executor_v<decltype(exec)> == true);
+    CHECK(execution::is_executor_v<int> == false);
+  }
+}
+
+TEST_CASE_TEMPLATE_DEFINE("Property Traits ", E, property_traits) {
+  auto exec = E{};
+
+  SUBCASE("can_require") {
+    CHECK(execution::can_require_v<decltype(exec), blocking_t::always_t> ==
+          true);
+    CHECK(execution::can_require_v<decltype(exec), blocking_t::never_t> ==
+          false);
+  }
+
+  SUBCASE("can_prefer") {
+    CHECK(execution::can_prefer_v<decltype(exec), blocking_t::always_t> ==
+          true);
+    CHECK(execution::can_prefer_v<decltype(exec), blocking_t::never_t> ==
+          false);
+  }
+}
+
+TEST_CASE_TEMPLATE_DEFINE("Properties", E, properties) {
```

Comment:
`E` is not used anywhere, which likely means this shouldn't be a `TEST_CASE_TEMPLATE_DEFINE`

Reply:
Resolved in #12